src/dynamic_programming/python/can_sum/src/can_sum.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+# Python Version: 3.12
+"""
+File: can_sum.py
+Author: Vlad421
+Date: 1/20/2024
+Description: Checks if array could produce given sum
+"""
+
+
+def can_sum_basic(number: int, arr: list[int]) -> bool:
+    """Basic brute force algorythm
+
+    Parameters:
+        arr: list[int] - List of items to sum
+        number:int - Target sum
+
+    Returns -> bool    
+
+    >>> can_sum_basic([2,1],1)
+    True """
+    if number < 0:
+        return False
+    elif number == 0:
+        return True
+
+    for i in arr:
+        reminder = number-i
+
+        return can_sum_basic(reminder, arr[1:]) or can_sum_basic(number, arr[1:])
+    return False
+
+
+def can_sum_memo(number: int, arr: list[int], __memo: int = 0) -> bool:
+    """Algorythm with memorization
+
+    Parameters:
+        arr: list[int] - List of items to sum
+        number:int - Target sum
+        memo:int - memorization for recurcive call only
+
+    Returns -> bool    
+
+    >>> can_sum_memo([3,1,1],2)
+    True """
+
```

Comment:
Maybe comments will help to understang how this recursion is working

Reply:
pretty muck same as previous but use memo

Reply:
the pervios (without the memo) also lack comments